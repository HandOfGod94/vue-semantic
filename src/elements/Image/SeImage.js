import SeImageComponent from './SeImage.functional'

export default {
  name: 'se-image',
  props: {
    as: { type: String, default: 'img' },
    src: { type: String, require: true },
    size: String,
    hidden: Boolean,
    disabled: Boolean,
    avatar: Boolean,
    bordered: Boolean,
    fluid: Boolean,
    rounded: Boolean,
    circular: Boolean,
    verticalAlign: String,
    centered: Boolean,
    spaced: [Boolean, String],
    float: String
  },

  computed: {
    classObject () {
      return {
        ui: true,
        hidden: this.hidden,
        disabled: this.disabled,
        [this.size]: this.size,
        bordered: this.bordered,
        avatar: this.avatar,
        fluid: this.fluid,
        rounded: this.rounded,
        circular: this.circular,
        [`${this.verticalAlign} aligned`]: this.verticalAlign,
        centered: this.centered,
        [`${this.spaced} spaced`]: !!this.spaced,
        [`${this.float} floated`]: this.float,
        image: true
      }
    }
  },

  render (createElement) {
    return createElement(SeImageComponent, {
      attrs: {
        ...this.$props,
        slots: this.$slots
      },
      class: this.classObject
    })
  }
}
