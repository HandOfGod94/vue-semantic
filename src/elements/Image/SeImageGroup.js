import Vue from 'vue'

export default {
  name: 'se-image-group',
  props: {
    size: String
  },
  computed: {
    classes: function () {
      let classes = {
        ui: true,
        [this.size]: this.size,
        images: true
      }
      return classes
    }
  },
  render: function (h) {
    return (
      <div class={this.classes}>
        {this.$slots.default}
      </div>
    )
  }
}
