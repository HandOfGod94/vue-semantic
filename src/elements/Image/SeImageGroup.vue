<template>
  <div :class="classObject">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'se-image-group',
  props: {
    size: String
  },
  computed: {
    classObject () {
      return {
        ui: true,
        [this.size]: this.size,
        images: true
      }
    }
  }
}
</script>
