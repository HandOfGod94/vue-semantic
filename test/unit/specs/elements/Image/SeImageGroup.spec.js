import SeImage from '@/elements/Image/SeImage'
import SeImageGroup from '@/elements/Image/SeImageGroup'
import { mount } from '@vue/test-utils'

const images = `
  <template>
    <se-image src="static/user.png" />
    <se-image src="static/user.png" disabled />
    <se-image src="static/user.png" rounded />
  </template>
`

describe('SeImageGroup snapshot test', () => {
  it('should match with existing snapshot', () => {
    const data = {
      propsData: {
        size: 'small'
      },
      slots:
      {
        default: images
      },
      stubs: {
        'se-image': SeImage
      }
    }
    const wrapper = mount(SeImageGroup, data)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
