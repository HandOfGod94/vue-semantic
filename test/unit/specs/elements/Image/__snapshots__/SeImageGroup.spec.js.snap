// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeImageGroup snapshot test should match with existing snapshot 1`] = `
<div class="ui small images">
    <img src="static/user.png" class="ui image">
    <img src="static/user.png" class="ui disabled image">
    <img src="static/user.png" class="ui rounded image">
</div>
`;
